test/apps/iodemo/io_demo.cc

```diff
@@ -597,45 +628,51 @@ class P2pDemoCommon : public UcxContext {
         ucs_memory_type_t        _memory_type;
         UcxContext               *_map_context;
         ucp_mem_h                _memh;
+        bool                     _zcopy;
     };
 
     class BufferIov {
     public:
         BufferIov(size_t size, MemoryPool<BufferIov> &pool) :
                 _data_size(0lu), _memory_type(UCS_MEMORY_TYPE_UNKNOWN),
-                _validate(false), _pool(pool), _extra_buf(NULL)
+                _validate(false), _pool(pool)
         {
             _iov.reserve(size);
         }
 
         size_t size() const
         {
-            assert(!_iov.empty() || _extra_buf);
-            return _iov.size() + !!_extra_buf;
+            assert(!_iov.empty());
+            return _iov.size();
         }
 
         size_t data_size() const
         {
-            assert(!_iov.empty() || _extra_buf);
+            assert(!_iov.empty());
             return _data_size;
         }
 
         ucs_memory_type_t mem_type() const
         {
-            assert(!_iov.empty() || _extra_buf);
+            assert(!_iov.empty());
             return _memory_type;
         }
 
         void init(size_t data_size, BufferMemoryPool<Buffer> &chunk_pool,
-                  uint32_t sn, uint64_t conn_id, bool validate)
+                  uint32_t sn, uint64_t conn_id, bool validate,
+                  void *zcopy_buf = NULL)
         {
             assert(_iov.empty());
-            assert(_extra_buf == NULL);
 
             _validate     = validate;
             _data_size    = data_size;
             _memory_type  = chunk_pool.memory_type();
             Buffer *chunk = chunk_pool.get();
+            if (zcopy_buf != NULL) {
+                chunk->set_buffer(zcopy_buf);
+                chunk->set_capacity(_data_size);
+                chunk->resize(_data_size);
```

Comment:
```suggestion
                chunk->set_buffer(zcopy_buf, _data_size);
```
everything else should be hidden from user